sale_ux/models/sale_order.py

```diff
@@ -22,6 +22,13 @@ class SaleOrder(models.Model):
         track_visibility='onchange',
         copy=False,
     )
+    commercial_partner_id = fields.Many2one(
+        'res.partner',
+        string='Commercial Entity',
+        related='partner_id.commercial_partner_id',
+        store=True,
```

Comment:
Dont need to store the related field to make it searchable,
Only need to to add a search method instead: this way we are saving a lot of database space

```diff
- store=True,
+ search='_search_commercial_partner_id',
```

The search method is like:

```python
     @api.model
     def _search_commercial_partner_id(self, operator, operand):
         return [('partner_id.commercial_partner_id', operator, operand)]
```

Reply:
To be searchable, if I'm not wrong, related fields don't need "search", they are searchable by default. From v9+ I think that you can also group by a related field without making it stored, So I beleieve that just removing  store=True is enaught (no need for search method)